tests/validation/tests/v3_api/test_hosted_eks_cluster.py

```diff
@@ -0,0 +1,91 @@
+import os
+from .common import get_user_client, validate_cluster, \
+    cluster_cleanup, random_test_name
+import pytest
+
+EKS_ACCESS_KEY = os.environ.get('RANCHER_EKS_ACCESS_KEY', "")
+EKS_SECRET_KEY = os.environ.get('RANCHER_EKS_SECRET_KEY', "")
+EKS_REGION = os.environ.get('RANCHER_EKS_REGION', "us-east-2")
+EKS_K8S_VERSION = os.environ.get('RANCHER_EKS_K8S_VERSION', "1.15")
+KMS_KEY = os.environ.get('RANCHER_KMS_KEY', None)
+SECRET_ENCRYPTION = os.environ.get('RANCHER_SECRET_ENCRYPTION', False)
+LOGGING_TYPES = os.environ.get('RANCHER_LOGGING_TYPES', None)
+ekscredential = pytest.mark.skipif(not (EKS_ACCESS_KEY and EKS_SECRET_KEY),
+                                   reason='EKS Credentials not provided, '
+                                          'cannot create cluster')
+
+@ekscredential
+def test_create_hosted_eks_cluster():
+    client, cluster = create_and_validate_eks_cluster(EKS_K8S_VERSION)
+    cluster_cleanup(client, cluster)
+
+
+def create_and_validate_eks_cluster(k8s_version):
+    client = get_user_client()
+    eks_config = get_eks_config(k8s_version)
+    print("Creating EKS cluster")
+    cluster = client.create_cluster(eks_config)
+    print(cluster)
+    cluster = validate_cluster(client, cluster, check_intermediate_state=True,
+                               skipIngresscheck=True)
+    return client, cluster
+
+
+def get_eks_config(version):
+    client = get_user_client()
+    ec2_cloud_credential_config = {
+        "accessKey": EKS_ACCESS_KEY,
+        "secretKey": EKS_SECRET_KEY
+    }
+    ec2_cloud_credential = client.create_cloud_credential(
+        amazonec2credentialConfig=ec2_cloud_credential_config
+    )
+    ec2_cloud_credential.id
+    cluster_name = random_test_name("test-auto-eks")
+    nodegroup_name = random_test_name("test-ng")
+    logging_types = []
+    if LOGGING_TYPES is not None:
+        temp = LOGGING_TYPES.split(",")
+        for logging in temp:
+            logging_types.append(logging)
+    else:
+        logging_types = None
+    eks_config = {
+        "amazonCredentialSecret": ec2_cloud_credential.id,
+        "displayName": cluster_name,
+        "imported": False,
+        "kmsKey": KMS_KEY,
+        "kubernetesVersion": version,
+        "privateAccess": False,
+        "publicAccess": True,
+        "region": EKS_REGION,
+        "secretsEncryption": SECRET_ENCRYPTION,
+        "subnets": [],
+        "tags": {},
+        "type": "eksclusterconfigspec",
+        "nodeGroups":[{
+            "desiredSize": 1,
+            "diskSize": 20,
+            "gpu": False,
+            "instanceType": "m5.large",
+            "maxSize": 1,
+            "minSize": 1,
```

Comment:
@sowmyav27 Can we have the node group size set to 2 ,so that inter host networking gets tested as part of validate_cluster() call.
Also can we validate that we did get 2 worker nodes created as part of the EKS cluster ?